src/lib/components/dialog/QDialog.svelte

```diff
@@ -19,41 +18,51 @@
   const emit = createEventDispatcher();
   let dialogElement: HTMLDialogElement;
 
-  $: canHideOnClickOutside = value === true && persistent !== true;
+  let opened = false;
+
+  onMount(() => {
+    opened = dialogElement.open;
+    opened ? (dialogElement.style.display = "block") : (dialogElement.style.display = "none");
+  });
+
+  $: canHideOnClickOutside = opened && persistent !== true;
 
   $: positionClass = ["top", "right", "bottom", "left"].includes(position!) ? position : undefined;
 
   $: classes = createClasses(
-    [value && "active", positionClass, modal && "modal", fullscreen && "max"],
+    [opened && "active", positionClass, modal && "modal", fullscreen && "max"],
     {
       component: "q-dialog",
       userClasses,
     }
   );
 
-  $: value === true
-    ? modal
-      ? dialogElement?.showModal()
-      : dialogElement?.show()
-    : dialogElement?.close();
-
   export function hide() {
-    if (value === true) {
-      value = false;
+    if (dialogElement && dialogElement.open) {
+      dialogElement.close();
+      opened = false;
+      setTimeout(() => {
```

Comment:
Why?

Reply:
To avoid being able to focus the elements within the dialog without having to change all tabindex (which can be custom by users)